lib/bracket-matcher.coffee

```diff
@@ -218,3 +224,6 @@ class BracketMatcher
 
   unsubscribe: ->
     @subscriptions.dispose()
+
+  getScopedSetting: (key) ->
+    atom.config.get(key, scope: @editor.getLastCursor().getScopeDescriptor())
```

Comment:
I actually think this would be a great addition to bracket-matcher, so thanks for adding it! :sparkles:  

I am trying to understand its consequences, and my main concern with the comment above was to handle only the last cursor scope when inserting text with many cursors (which _may_ have different scopes), but I just realized that we have [this line](https://github.com/atom/bracket-matcher/blob/master/lib/bracket-matcher.coffee#L52-L53) which prevents closing brackets when there's a multi-cursor edit. 

Still, this could create an inconsistent behavior when `@wrapSelectionInBrackets` gets called. Specifically, if we consider only the last cursor, we might end up wrapping some selections that, because of the scoped config, the user didn't want to wrap, or vice versa. At the same time, fetching the config for each cursor scope descriptor might cause some serious slowdowns when we have lots of them, and feels suboptimal.

Pragmatically, I think we could proceed like the following:

1. Capture a CPU profile of wrapping 1000 selections with brackets (the **before profile**);
2. Try to fetch the cursor scope descriptor for each selection (`selection.cursor.getScopeDescriptor()`);
3. Capture a CPU profile of wrapping 1000 selections with brackets (the **after profile**);
4. If the two profiles are similar in terms of performance, then we should definitely favor correctness, and keep 2);
5. Otherwise, it might be acceptable to use only the latest cursor scope descriptor, and keep the code as it is (and as we are already doing [for this line](https://github.com/atom/bracket-matcher/blob/master/lib/bracket-matcher.coffee#L196)).

What do you think, @deprint? Please, let me know if I can provide additional guidance on the profiling stuff or anything else related to the points above. Thanks! :fireworks: